build/helper/codegen_helper.py

```diff
@@ -1401,34 +1398,40 @@ def get_enum_value_snippet(value):
 ]
 
 
-def test_get_method_return_snippet_vi():
+def test_get_library_interpreter_method_return_snippet_vi():
     param = [parameters_for_testing[0]]
-    assert get_method_return_snippet(param, config_for_testing) == 'return'
+    assert get_library_interpreter_method_return_snippet(param, config_for_testing) == 'return'
 
 
-def test_get_method_return_snippet_int():
+def test_get_library_interpreter_method_return_snippet_int():
     param = [parameters_for_testing[1]]
-    assert get_method_return_snippet(param, config_for_testing) == 'return int(output_ctype.value)'
+    assert get_library_interpreter_method_return_snippet(param, config_for_testing) == 'return int(output_ctype.value)'
 
 
-def test_get_method_return_snippet_string():
+def test_get_library_interpreter_method_return_snippet_string():
     param = [parameters_for_testing[2]]
-    assert get_method_return_snippet(param, config_for_testing) == 'return error_message_ctype.value.decode(self._encoding)'
+    assert get_library_interpreter_method_return_snippet(param, config_for_testing) == 'return error_message_ctype.value.decode(self._encoding)'
 
 
-def test_get_method_return_snippet_custom_type():
+def test_get_library_interpreter_method_return_snippet_custom_type():
     param = [parameters_for_testing[3]]
-    assert get_method_return_snippet(param, config_for_testing) == 'return [custom_struct.CustomStruct(array_out_ctype[i]) for i in range(self.get_array_size_for_python_code())]'
+    assert get_library_interpreter_method_return_snippet(param, config_for_testing) == 'return [custom_struct.CustomStruct(array_out_ctype[i]) for i in range(self.get_array_size_for_python_code())]'
 
 
-def test_get_method_return_snippet_enum():
+def test_get_library_interpreter_method_return_snippet_enum():
     param = [parameters_for_testing[4], parameters_for_testing[5]]
-    assert get_method_return_snippet(param, config_for_testing) == 'return [enums.Turtle(an_array_ctype[i]) for i in range(number_of_elements_ctype.value)]'
+    assert get_library_interpreter_method_return_snippet(param, config_for_testing) == 'return [enums.Turtle(an_array_ctype[i]) for i in range(number_of_elements_ctype.value)]'
 
 
-def test_get_method_return_snippet_into():
+def test_get_library_interpreter_method_return_snippet_into():
     param = [parameters_for_testing[4], parameters_for_testing[7]]
-    assert get_method_return_snippet(param, config_for_testing, use_numpy_array=True) == 'return'
+    assert get_library_interpreter_method_return_snippet(param, config_for_testing, use_numpy_array=True) == 'return'
+
+
+def test_get_session_method_return_snippet():
```

Comment:
I think you may need more test cases for 'get_session_method_return_snippet'. You may have mostly copied the code from 'get_library_interpreter_method_return_snippet' but now that they are separate methods you can't count on testing for that method to make sure all of the behavior for this method is correct.

Reply:
Before this change, the get_method_return_snippet function had a matrix of [ctypes_conversions] x [converters_conversions]. Now, the library_interpreter one has just [ctypes_conversions], and the session one has just [converters_conversions]. When unit-testing the session one, I shouldn't need to test any of the ctypes_conversions, and since the _converters API is the same for all converter types, I just need one unit test for it.

The overall flow still has a matrix of [ctypes_conversions] x [converters_conversions], because the Session's conversions get layered on top of the LibraryInterpreter's conversions, but that's now the job of system tests, not unit tests.

Reply:
Gotcha. I'll check on that today.

Reply:
Added tests for tests for use_numpy_array=True/False.